logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -188,6 +194,7 @@ public void open() throws IOException {
                 } else {
                     pageIO.open(cp.getMinSeqNum(), cp.getElementCount());
                     addTailPage(cp, PageFactory.newTailPage(cp, this, pageIO));
+                    diskNeeded -= (long)pageIO.getHead();
```

Comment:
for my own clarification:

Since we opened this PageIO with the checkpoint's element count, we skipped over that many items moving the head to the end of the file, so `PageIO#getHead()` should effectively return the filesize in bytes.

We are then subtracting the size of all existing non-fully-ack'd files from our `maxBytes` configuration; when we are done looking at all the pages, `diskNeeded` will represent the _remaining_ amount of disk space we're allowed to consume.

In this light, what we're doing makes sense.

Reply:
+1